### PR TITLE
Add Faucet page

### PR DESCRIPTION
**Pull request by evilpeach**
## Describe your changes
- add Faucet for requesting Testnet OSMO (Testnet only)

### CODE REVIEW

File: src/lib/styles/theme/colors.ts [hunk 1]
```diff
@@ -25,16 +25,19 @@ const extendedColors: BasicColor = {
     main: "#FF666E",
     light: "#FF8086",
     dark: "#b43e44",
+    background: "#4C1A1D",
   },
   warning: {
     main: "#FFBB33",
     light: "#FFCC66",
     dark: "#CC8800",
+    background: "523600",
```
Comment (bkioshn) [inline src/lib/styles/theme/colors.ts:34]: "#"

File: src/lib/pages/faucet/index.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    <WasmPageContainer>
+      <BackButton alignSelf="flex-start" />
+      <Heading as="h5" variant="h5">
+        Testnet OSMO Faucet
+      </Heading>
+      <Text variant="body2" color="gray.100" pt={4} textAlign="center" mb={8}>
+        The Faucet provides 1.00 testnet OSMO per request for osmo-test-4
+        network. Please note that there is a limit of one request per hour.
+      </Text>
+      <TextInput
```
Comment (bkioshn) [inline src/lib/pages/faucet/index.tsx:155]: Should we add placeholder?

File: src/lib/pages/faucet/index.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      });
+  };
+
+  const disabled = status.state !== "success" || isLoading;
+
+  return (
+    <WasmPageContainer>
+      <BackButton alignSelf="flex-start" />
+      <Heading as="h5" variant="h5">
+        Testnet OSMO Faucet
```
Comment (bkioshn) [inline src/lib/pages/faucet/index.tsx:149]: OSMO Testnet Faucet

File: src/lib/components/button/FaucetButton.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    <Button
+      gap={2}
+      onClick={onClick}
+      borderWidth="1px"
+      borderColor="gray.600"
+      background="unset"
+    >
+      {/* TODO: replace icon and styles later */}
+      <Icon as={MdLink} boxSize={4} />
+      <Text color="gray.600" fontWeight={700}>
```
Comment (bkioshn) [inline src/lib/components/button/FaucetButton.tsx:28]: should be text.dark

File: src/lib/components/button/FaucetButton.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      pathname: "/faucet",
+    });
+  };
+
+  return isTestnet ? (
+    <Button
+      gap={2}
+      onClick={onClick}
+      borderWidth="1px"
+      borderColor="gray.600"
```
Comment (bkioshn) [inline src/lib/components/button/FaucetButton.tsx:23]: pebble.600

File: src/lib/pages/faucet/index.tsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+      </Heading>
+      <Text variant="body2" color="gray.100" pt={4} textAlign="center" mb={8}>
```
Comment (bkioshn) [inline src/lib/pages/faucet/index.tsx:151]: pebble.600

**Review by bkioshn — CHANGES_REQUESTED** [review]

File: src/lib/pages/faucet/index.tsx (hunk 3 above)
Comment (evilpeach) [reply src/lib/pages/faucet/index.tsx:149]: Testnet OSMO Faucet is correct.

File: src/lib/pages/faucet/index.tsx (hunk 2 above)
Comment (evilpeach) [reply src/lib/pages/faucet/index.tsx:153]: I'll discuss with designer.

File: src/lib/pages/faucet/index.tsx [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+      });
+  };
+
+  const disabled = status.state !== "success" || isLoading;
+
+  return (
+    <WasmPageContainer>
+      <BackButton alignSelf="flex-start" />
+      <Heading as="h5" variant="h5">
+        Testnet OSMO Faucet
```
Comment (poomthiti) [inline src/lib/pages/faucet/index.tsx:147]: todo: handle token symbol by chain

File: src/lib/pages/faucet/index.tsx [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+        }
+      />
+      <Button
+        mt={8}
+        w="full"
+        onClick={onSubmit}
+        isLoading={isLoading}
+        disabled={disabled}
+      >
+        Receive 10.00 Testnet OSMO
```
Comment (poomthiti) [inline src/lib/pages/faucet/index.tsx:183]: osmo ditto

**Review by poomthiti — CHANGES_REQUESTED** [review]

**Review by poomthiti — APPROVED** [review]

**Review by bkioshn — APPROVED** [review]

**Pull request merged**